span.go

```diff
@@ -346,6 +348,13 @@ func (s *Span) End() {
 	}
 	if s.Outcome == "" {
 		s.Outcome = s.Context.outcome()
+		if s.Outcome == "" {
+			if s.errorCaptured {
```

Comment:
nice, this access doesn't need to be coordinated because the writer acquires a `s.mu.Rlock` and modifications won't be allowed to happen since this function holds `s.mu.Lock`

Reply:
Yup :) Maybe it would be worth commenting on the locks when they should be held. The SpanData/TransactionData locks only need to be held when holding an RLock on the Span/Transaction lock.